H2MLauncher.UI/Dialog/Views/PasswordDialog.xaml.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Media;
+
+using H2MLauncher.UI.Dialog;
+
+namespace H2MLauncher.UI
+{
+    public partial class PasswordDialog : UserControl
+    {
+        public PasswordDialog()
+        {
+            InitializeComponent();
+
+        }
+        private void OK_Click(object sender, RoutedEventArgs e)
+        {
+            if (this.DataContext is not PasswordViewModel viewModel)
+                return;
+
+            viewModel.Password = PasswordInput.Password;
+
+            Window.GetWindow(this).DialogResult = true;
```

Comment:
Please use the command on the view model (`viewModel.CloseCommand.Execute(true)`)